### PR TITLE
Cleanup redundant failure reason in InterPodAffinity plugin

### PR DESCRIPTION
**Pull request by chendave**
Both `ErrReasonAffinityRulesNotMatch` and `ErrReasonAntiAffinityRulesNotMatch` are
more precise than `ErrReasonAffinityNotMatch`.

Signed-off-by: Dave Chen <dave.chen@arm.com>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ahg-g** [comment]
I think this was a legacy thing. The argument for having this extra common message was that someone processing the logs may want to match against a common substring for all affinity related errors. I would remove it though, I think people who want to do this can still do this using a regex. 

However, since this appears in events, may be we need to maintain backward compatibility of the message?

**Comment by chendave** [comment]
> I would remove it though, I think people who want to do this can still do this using a regex.

+1, I think downstream should adapt to the upstream.

> we need to maintain backward compatibility of the message

I don't know if there is any external dependency on this, if we need to maintain backward compatibility, should us keep it as unchanged? any suggestion?

**Comment by chendave** [comment]
race detected in the testcases, not sure whether sig storage are working on this.

> I0602 04:09:05.203186   58395 shared_informer.go:247] Caches are synced for attach detach 
    testing.go:1092: race detected during execution of test
    --- FAIL: Test_ADC_VolumeAttachmentRecovery/Deleted_Pod_with_migrated_PV (0.11s)


/retest

**Comment by yuzhiquan** [comment]
> race detected in the testcases, not sure whether sig storage are working on this.
> 
> > I0602 04:09:05.203186   58395 shared_informer.go:247] Caches are synced for attach detach
> > testing.go:1092: race detected during execution of test
> > --- FAIL: Test_ADC_VolumeAttachmentRecovery/Deleted_Pod_with_migrated_PV (0.11s)
> 
> /retest

Meet this data race too, and make fix try in https://github.com/kubernetes/kubernetes/pull/102477, waiting more review.

**Comment by ahg-g** [comment]
> I don't know if there is any external dependency on this, if we need to maintain backward compatibility, should us keep it as unchanged? any suggestion?

What problem are you trying to solve with this PR?

**Comment by alculquicondor** [comment]
> However, since this appears in events, may be we need to maintain backward compatibility of the message?

I don't think so. We have changed them in the past, for example to add specific unmatched taints. I would be more careful with the reason.

I'm in favor of removing this redundant reason.

**Comment by chendave** [comment]
> What problem are you trying to solve with this PR?

It just a cleanup, found this string doesn't provide any extra information, no problem to keep it as is.

But if we decide to keep it as is we'd better add some comments there to explain why we need this common extra string.

**Comment by chendave** [comment]
> I'm in favor of removing this redundant reason.

we have some testcase with either / or logic on the status code (affinity or anti-affinity will return different code) , remove this extra string will at least make the testcase more straightforward.

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go [hunk 1]
```diff
@@ -1344,27 +1344,19 @@ func TestNodesWherePreemptionMightHelp(t *testing.T) {
...
 			nodesStatuses: framework.NodeToStatusMap{
-				"node1": framework.NewStatus(framework.Unschedulable, interpodaffinity.ErrReasonAffinityNotMatch),
+				"node1": framework.NewStatus(framework.Unschedulable, interpodaffinity.ErrReasonAntiAffinityRulesNotMatch),
 				"node2": framework.NewStatus(framework.UnschedulableAndUnresolvable, nodename.ErrReason),
 				"node3": framework.NewStatus(framework.UnschedulableAndUnresolvable, nodeunschedulable.ErrReasonUnschedulable),
 			},
 			expected: sets.NewString("node1", "node4"),
 		},
 		{
-			name: "pod with both pod affinity and anti-affinity should be tried",
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1356]: why was this test removed?

**Comment by ahg-g** [comment]
sg, lets remove then, I left one comment.

/lgtm
/approve
/hold

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go (hunk 1 above)
Comment (chendave) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1356]: the name of the testcase itself is not correct, only the anti-affinity should be tried which has been covered by the testcase above, `ErrReasonAntiAffinityRulesNotMatch should be tried as it indicates that the pod...`

**Pull request merged**